components/sync15/src/status.rs

```diff
@@ -0,0 +1,59 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::error::{Error, ErrorKind, StorageHttpError};
+
+/// The general status of sync - should probably be moved to the "sync manager"
+/// once we have one!
+#[derive(Debug, PartialEq)]
+pub enum ServiceStatus {
+    /// Everything is fine.
+    Ok,
+    /// Some general network issue.
+    NetworkError,
+    /// Some apparent issue with the servers.
+    ServiceError,
+    /// Some external FxA action needs to be taken.
+    AuthenticationError,
+    /// We declined to do anything for backoff or rate-limiting reasons.
+    BackedOff,
+    /// We were interrupted.
+    Interrupted,
+    /// Something else - you need to check the logs for more details. May
+    /// or may not be transient, we really don't know.
+    OtherError,
+}
+
+impl ServiceStatus {
+    // This is a bit naive and probably will not survive in this form in the
+    // SyncManager - eg, we'll want to handle backoff etc.
+    pub fn from_err(err: &Error) -> ServiceStatus {
```

Comment:
It looks like you already have an owning reference to `err` everywhere you call this...could this be `impl From<Error> for ServiceStatus`, or `impl From<&Error> for ServiceStatus`?

Reply:
(In case you try to do `From<&Error>`, it would need to be `impl<'a> From<&'a Error>` (I only mention it since it could cause someone to get stuck trying to figure out the right syntax for a while)

Reply:
I don't think it will survive in this form anyway, so I'm not bothering with this at this time.